### PR TITLE
refactor(database): update neo4j to 4.4

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

This PR updates the neo4j database to 4.4

### Issues

- fixes https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/6087
- based on https://github.com/Ocelot-Social-Community/Ocelot-Social/pull/6303

### Todo

- [ ] Investigate how this behaves on a kubernetes cluster

### How to test?

1. check current database contents
2. check out branch
3. `docker-compose up --build`
4. See all the old content
5. ...
6. Profit

### CODE REVIEW

File: backend/src/middleware/notifications/notificationsMiddleware.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const queryNotificationEmails = async (context, notificationUserIds) => {
     RETURN emailAddress {.email}
   `
   const session = context.driver.session()
-  const writeTxResultPromise = session.writeTransaction(async (transaction) => {
+  const writeTxResultPromise = session.readTransaction(async (transaction) => {
```
Comment (Tirokk) [inline backend/src/middleware/notifications/notificationsMiddleware.js:18]: Cool catch 👍🏼

File: backend/src/schema/resolvers/moderation.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ export default {
             MATCH (resource {id: $params.resourceId})<-[:BELONGS_TO]-(report:Report {closed: false})
             WHERE resource:User OR resource:Post OR resource:Comment
             MERGE (report)<-[review:REVIEWED]-(moderator)
-            ON CREATE SET review.createdAt = $dateTime, review.updatedAt = review.createdAt
+            ON CREATE SET review.createdAt = $dateTime, review.updatedAt = $dateTime
```
Comment (Tirokk) [inline backend/src/schema/resolvers/moderation.js:15]: We did this to be really sure to have the same date on both on creation.

**Review by Tirokk — COMMENTED** [review]
For me that all looks nice as prereview.
I haven't tested this locally now.

But we really have to investigate how that works on real data.
So maybe we should test this with real data on staging. 👍🏼

File: backend/src/schema/resolvers/moderation.js (hunk 2 above)
Comment (ulfgebhardt) [reply backend/src/schema/resolvers/moderation.js:15]: This doesn't work with 4.4 and will make the query fail. Since its a variable it is the same value.

**Comment by ulfgebhardt** [comment]
> For me that all looks nice as prereview. heart_eyespartying_face I haven't tested this locally now.
> 
> But we really have to investigate how that works on real data. So maybe we should test this with real data on staging. 👍🏼

Please test this.

**Review by Elweyn — CHANGES_REQUESTED** [review]

**Review by Elweyn — CHANGES_REQUESTED** [review]
image

**Review by Elweyn — DISMISSED** [review]

**Comment by ulfgebhardt** [comment]
Reporting seems not to work correctly?

**Comment by ulfgebhardt** [comment]
And its actually detected by our fullstack tests ❤️ 
image

**Comment by ulfgebhardt** [comment]
@Elweyn Please review again

**Review by Elweyn — APPROVED** [review]
Works fine for me!
- Create Article 
- Create Event
- Create Group
- Reports
Thanks for updating this

**Pull request merged**